var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -298,13 +299,5 @@ def test(self):
             exe = os.path.join(self.prefix.bin, test[0])
             args = test[1:]
             reason = 'test {0}: "{1}"'.format(i, " ".join(test))
-            self.run_test(
-                exe,
-                args,
-                [],
-                0,
-                installed=False,
-                purpose=reason,
-                skip_missing=True,
-                work_dir=self.prefix.bin,
-            )
+            exe = which(exe)
```

Comment:
This is not being passed any of the `args`.

Reply:
Also, it should be using the `test_parts()` method in order to get the equivalent failure handling that was in `run_test()`.

Reply:
The current docs include an example at https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-stand-alone-tests.

A rendering of the updated docs for that section should be at https://spack--44755.org.readthedocs.build/en/44755/packaging_guide.html#adding-stand-alone-test-parts.